Packs/MailListener/Integrations/MailListenerV2/MailListenerV2.py

```diff
@@ -0,0 +1,220 @@
+            'rawHeaders': self.headers
+        }
+        if self.html:
+            raw_json['html'] = self.html
+        if self.raw_body:
+            raw_json['rawBody'] = self.raw_body
+        return raw_json
+
+
+def fetch_incidents(client: IMAPClient,
+                    last_run: dict,
+                    first_fetch_time: str,
+                    include_raw_body: bool,
+                    permitted_from_addresses: str,
+                    permitted_from_domains: str,
+                    delete_processed: bool,
+                    limit: int
+                    ):
+    """
+    This function will execute each interval (default is 1 minute).
+
+    Args:
+        client: HelloWorld client
+        last_run: The greatest incident created_time we fetched from last fetch
+        first_fetch_time: If last_run is None then fetch all incidents since first_fetch_time
+        include_raw_body: Whether to include the raw body of the mail in the incident's body
+        permitted_from_addresses: A string representation of list of mail addresses to fetch from
+        permitted_from_domains: A string representation list of domains to fetch from
+        delete_processed: Whether to delete processed mails
+        limit: The maximum number of incidents to fetch each time
+
+    Returns:
+        next_run: This will be last_run in the next fetch-incidents
+        incidents: Incidents that will be created in Demisto
+    """
+    # Get the last fetch time, if exists
+    last_fetch = last_run.get('last_fetch')
+
+    # Handle first time fetch
+    if last_fetch is None:
+        latest_created_time = parse(f'{first_fetch_time} UTC')
+    else:
+        latest_created_time = datetime.fromisoformat(last_fetch)
+    messages_query = generate_search_query(latest_created_time, permitted_from_addresses, permitted_from_domains)
+    messages = client.search(messages_query)
+    messages = messages[:limit]
+    mails_fetched = []
+    for message_data in client.fetch(messages, 'RFC822').values():
+        message_bytes = message_data.get(b'RFC822')
+        if not message_bytes:
+            continue
+        email_message = parse_from_bytes(message_bytes)
+        # The search query filters emails by day, not by exact date
+        email_message_object = Email(email_message, include_raw_body)
+        if email_message_object.date > latest_created_time:
+            mails_fetched.append(email_message_object)
+    if mails_fetched:
+        latest_created_time = max(mails_fetched, key=lambda x: x.date).date
+    incidents = [mail.convert_to_incident() for mail in mails_fetched]
+    next_run = {'last_fetch': latest_created_time.isoformat()}
+    if delete_processed:
+        client.delete_messages(messages)
+    return next_run, incidents
+
+
+def generate_search_query(latest_created_time: datetime, permitted_from_addresses: str, permitted_from_domains: str):
+    """
```

Comment:
please add to the docstring what the function does, an example, and fill in the return section